packages/liveblocks-node/src/client.ts

```diff
@@ -84,6 +86,66 @@ type RoomInfoOriginal = Omit<RoomInfo, "lastConnectionAt" | "createdAt"> & {
   createdAt?: string;
 };
 
+type CommentReactionUser =
+  CommentDataOriginal["reactions"][number]["users"][number];
+
+/**
+ * This CommentData type modifies the (original) CommentData type from @liveblocks/core (which is also the shape of the comment data returned by the API)
+ * All properties that are dates are converted from ISO strings to Date objects.
+ */
+export type CommentData = Omit<
+  CommentDataOriginal,
+  "createdAt" | "editedAt" | "body" | "deletedAt" | "reactions"
+> & {
+  createdAt: Date;
+  editedAt?: Date;
+  reactions: {
+    emoji: string;
+    createdAt: Date;
+    users: CommentReactionUser[];
+  }[];
+} & (
+    | {
+        body: CommentBody;
+        deletedAt?: never;
+      }
+    | {
+        body?: never;
+        deletedAt: Date;
+      }
+  );
```

Comment:
This type setup seems a bit hard to maintain to me. I have a couple of suggestions to make maintaining this over time easier. Could we "flip this around", i.e. making this new `CommentData` (with possible Date values) the rich base type, and deriving the `CommentDataOriginal` (with strings) from it? That way, you go from a richer to a less rich type, which can be done using a generic transformation function that converts `Date` types to `string` types. The big benefit is that those types are more readable _and_ they don't require the maintenance of keeping the field union up to date, any time a new Date field would get added or removed. The same maintenance currently lies on the optional-or-required field information (on L100-101).

Suggestion:

- In `@liveblocks/core`, expose `CommentData` as a rich type that has `Date` values.
- In `@liveblocks/core`, also expose `CommentDataPlain`, which is defined as `CommentDataPlain = DatesToStrings<CommentData>`.
- Use `CommentDataPlain` in our Node endpoint responses.
- Use `CommentData` (with Dates) in our client's REST API responses.

Here's a TypeScript playground with an example of such helper type:

https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBDAnmYcBSBnCA7OBedLbAZQGMBDAG3KjgB9CcBBKKcxex7AeQCMArYKRgBuAFChIsBMlSYcZKjXxwMMKAEtsAc07YArgFtewWg14QIlYOVwMDlSuMnR4SFFxZsOBedgDaALrO4K4yHn58gsIqAN5w-gDWwIgAXKrqWtqB6X6c+tgAJsAAZlrAhXAAvuJiAPR1cAC0La1t7R2dXd09XWJi7qgAIuQwwBgAKhDEmToYADwTAHwqE3CgY0UYcCNjcAD8GZo6cOnx-gDScFpwyYgQJXATOTuj41Mzx9oLE5eBKzV+oM4AAlDSkAAWcTEcFhcFIUBsY0KTBg6V2wHEcLgxWsyNR+3RbyxcMR5EKOEoHGw42R6TJFOwVLgsWwEAA7kS9vZ9I4RNUgiTYdhyIZgOk1F8hddCukDMZTNLyNpgIS4PKTFBpWM1GrJVlOPqdEE9LynGIqkDZHAAAqWWgEDGTaazb7zMGQpbiBpwgB6+zEQA